### PR TITLE
document open GC registry

### PR DESCRIPTION
**Pull request by rainers**
documentation for https://github.com/dlang/druntime/pull/1924

### CODE REVIEW

File: spec/garbage.dd [hunk 1]
```diff
@@ -519,6 +519,46 @@ $(H2 $(LNAME2 precise_dataseg, Precise Scanning of the DATA and TLS segment))
...
+    $(P The new GC is added to the list of available garbage collectors that
+        can be selected via the usual configuration options, e.g. by embedding
+        `rt_options` into the binary:)
+        ---
+        extern (C) __gshared string[] rt_options = ["gcopt=gc:mygc"];
+        ---
+
+    $(P You can also remove the standard GC implementation from a statically
+        linked binary by redefining the function `extern(C) void* register_default_gcs()`.
+        This will cause all attempts to allocate GC managed memory to terminate
```
Comment (wilzbach) [inline spec/garbage.dd]: Maybe add sth. like "if no custom GC has been registered ..."

Comment (rainers) [reply spec/garbage.dd]: Ok, done.

**Comment by rainers** [comment]
Anything holding this back? The beta changelog already links to this documentation.

**Comment by thewilsonator** [comment]
Its labelled as waiting on a DMD PR. @wilzbach which one?

**Comment by rainers** [comment]
> Its labelled as waiting on a DMD PR. @wilzbach which one?

I guess that was meant for the druntime PR https://github.com/dlang/druntime/pull/1924

**Comment by ZombineDev** [comment]
Are we going to wait for https://github.com/dlang/druntime/pull/2491?

**Comment by wilzbach** [comment]
> Are we going to wait for [dlang/druntime#2491](https://github.com/dlang/druntime/pull/2491)?

2.085 will come with this feature, so we should at least document it (even if it's a bit harder to use).
Though once dlang/druntime#2491 is merged, we should probably update this page with the appropriate imports to `core.gc`

**Pull request merged**